### DISCUSSION TITLE
Getting the following error "Server Error (500)" when clicking the execute button

### DISCUSSION THREAD

**Discussion by ThivaS357** [frePPLe/frepple#497] [Q&A]
Oops, something went wrong.
Your administrator can find more information on the error in the log file /var/log/apache2/error.log

Hey Guys we are getting above error in our environment

please find apache2 error logs below

[Wed Feb 16 15:01:29.523659 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438] ERROR Internal Server Error: /execute/
[Wed Feb 16 15:01:29.523694 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438] Traceback (most recent call last):
[Wed Feb 16 15:01:29.523710 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/core/handlers/exception.py", line 34, in inner
[Wed Feb 16 15:01:29.523714 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     response = get_response(request)
[Wed Feb 16 15:01:29.523718 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/core/handlers/base.py", line 115, in _get_response
[Wed Feb 16 15:01:29.523722 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     response = self.process_exception_by_middleware(e, request)
[Wed Feb 16 15:01:29.523726 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/core/handlers/base.py", line 113, in _get_response
[Wed Feb 16 15:01:29.523730 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     response = wrapped_callback(request, *callback_args, **callback_kwargs)
[Wed Feb 16 15:01:29.523734 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/views/generic/base.py", line 71, in view
[Wed Feb 16 15:01:29.523738 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     return self.dispatch(request, *args, **kwargs)
[Wed Feb 16 15:01:29.523742 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/utils/decorators.py", line 45, in _wrapper
[Wed Feb 16 15:01:29.523746 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     return bound_method(*args, **kwargs)
[Wed Feb 16 15:01:29.523750 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/contrib/auth/decorators.py", line 21, in _wrapped_view
[Wed Feb 16 15:01:29.523754 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     return view_func(request, *args, **kwargs)
[Wed Feb 16 15:01:29.523757 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/utils/decorators.py", line 45, in _wrapper
[Wed Feb 16 15:01:29.523761 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     return bound_method(*args, **kwargs)
[Wed Feb 16 15:01:29.523765 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/utils/decorators.py", line 142, in _wrapped_view
[Wed Feb 16 15:01:29.523769 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     response = view_func(request, *args, **kwargs)
[Wed Feb 16 15:01:29.523773 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/lib/python3/dist-packages/freppledb/common/report.py", line 858, in dispatch
[Wed Feb 16 15:01:29.523777 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/lib/python3/dist-packages/freppledb/common/report.py", line 1792, in get
[Wed Feb 16 15:01:29.523781 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     )
[Wed Feb 16 15:01:29.523784 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/shortcuts.py", line 36, in render
[Wed Feb 16 15:01:29.523788 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     content = loader.render_to_string(template_name, context, request, using=using)
[Wed Feb 16 15:01:29.523792 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/loader.py", line 62, in render_to_string
[Wed Feb 16 15:01:29.523800 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     return template.render(context, request)
[Wed Feb 16 15:01:29.523804 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/backends/django.py", line 61, in render
[Wed Feb 16 15:01:29.523808 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     return self.template.render(context)
[Wed Feb 16 15:01:29.523812 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/base.py", line 171, in render
[Wed Feb 16 15:01:29.523816 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     return self._render(context)
[Wed Feb 16 15:01:29.523819 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/base.py", line 163, in _render
[Wed Feb 16 15:01:29.523823 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     return self.nodelist.render(context)
[Wed Feb 16 15:01:29.523827 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/base.py", line 937, in render
[Wed Feb 16 15:01:29.523831 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     bit = node.render_annotated(context)
[Wed Feb 16 15:01:29.523835 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/base.py", line 904, in render_annotated
[Wed Feb 16 15:01:29.523839 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     return self.render(context)
[Wed Feb 16 15:01:29.523842 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/loader_tags.py", line 150, in render
[Wed Feb 16 15:01:29.523846 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     return compiled_parent._render(context)
[Wed Feb 16 15:01:29.523850 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/base.py", line 163, in _render
[Wed Feb 16 15:01:29.523854 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     return self.nodelist.render(context)
[Wed Feb 16 15:01:29.523858 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/base.py", line 937, in render
[Wed Feb 16 15:01:29.523861 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     bit = node.render_annotated(context)
[Wed Feb 16 15:01:29.523865 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/base.py", line 904, in render_annotated
[Wed Feb 16 15:01:29.523869 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     return self.render(context)
[Wed Feb 16 15:01:29.523873 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/loader_tags.py", line 150, in render
[Wed Feb 16 15:01:29.523877 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     return compiled_parent._render(context)
[Wed Feb 16 15:01:29.523880 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/base.py", line 163, in _render
[Wed Feb 16 15:01:29.523884 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     return self.nodelist.render(context)
[Wed Feb 16 15:01:29.523891 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/base.py", line 937, in render
[Wed Feb 16 15:01:29.523895 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     bit = node.render_annotated(context)
[Wed Feb 16 15:01:29.523899 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/base.py", line 904, in render_annotated
[Wed Feb 16 15:01:29.523902 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     return self.render(context)
[Wed Feb 16 15:01:29.523951 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/loader_tags.py", line 150, in render
[Wed Feb 16 15:01:29.523957 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     return compiled_parent._render(context)
[Wed Feb 16 15:01:29.523961 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/base.py", line 163, in _render
[Wed Feb 16 15:01:29.523965 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     return self.nodelist.render(context)
[Wed Feb 16 15:01:29.523969 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/base.py", line 937, in render
[Wed Feb 16 15:01:29.523973 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     bit = node.render_annotated(context)
[Wed Feb 16 15:01:29.523977 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/base.py", line 904, in render_annotated
[Wed Feb 16 15:01:29.523980 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     return self.render(context)
[Wed Feb 16 15:01:29.523984 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/loader_tags.py", line 150, in render
[Wed Feb 16 15:01:29.523988 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     return compiled_parent._render(context)
[Wed Feb 16 15:01:29.523992 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/base.py", line 163, in _render
[Wed Feb 16 15:01:29.523996 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     return self.nodelist.render(context)
[Wed Feb 16 15:01:29.523999 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/base.py", line 937, in render
[Wed Feb 16 15:01:29.524003 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     bit = node.render_annotated(context)
[Wed Feb 16 15:01:29.524007 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/base.py", line 904, in render_annotated
[Wed Feb 16 15:01:29.524011 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     return self.render(context)
[Wed Feb 16 15:01:29.524015 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/loader_tags.py", line 62, in render
[Wed Feb 16 15:01:29.524018 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     result = block.nodelist.render(context)
[Wed Feb 16 15:01:29.524022 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/base.py", line 937, in render
[Wed Feb 16 15:01:29.524029 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     bit = node.render_annotated(context)
[Wed Feb 16 15:01:29.524033 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/base.py", line 904, in render_annotated
[Wed Feb 16 15:01:29.524037 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     return self.render(context)
[Wed Feb 16 15:01:29.524041 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/loader_tags.py", line 62, in render
[Wed Feb 16 15:01:29.524045 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     result = block.nodelist.render(context)
[Wed Feb 16 15:01:29.524048 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/base.py", line 937, in render
[Wed Feb 16 15:01:29.524052 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     bit = node.render_annotated(context)
[Wed Feb 16 15:01:29.524056 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/base.py", line 904, in render_annotated
[Wed Feb 16 15:01:29.524060 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     return self.render(context)
[Wed Feb 16 15:01:29.524064 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/defaulttags.py", line 309, in render
[Wed Feb 16 15:01:29.524067 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     return nodelist.render(context)
[Wed Feb 16 15:01:29.524071 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/base.py", line 937, in render
[Wed Feb 16 15:01:29.524075 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     bit = node.render_annotated(context)
[Wed Feb 16 15:01:29.524079 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/base.py", line 904, in render_annotated
[Wed Feb 16 15:01:29.524083 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     return self.render(context)
[Wed Feb 16 15:01:29.524086 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/defaulttags.py", line 209, in render
[Wed Feb 16 15:01:29.524090 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     nodelist.append(node.render_annotated(context))
[Wed Feb 16 15:01:29.524094 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/base.py", line 904, in render_annotated
[Wed Feb 16 15:01:29.524098 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     return self.render(context)
[Wed Feb 16 15:01:29.524102 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/local/lib/python3.8/dist-packages/django/template/library.py", line 192, in render
[Wed Feb 16 15:01:29.524106 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     output = self.func(*resolved_args, **resolved_kwargs)
[Wed Feb 16 15:01:29.524109 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/lib/python3/dist-packages/freppledb/common/templatetags.py", line 746, in call_method
[Wed Feb 16 15:01:29.524113 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     return method(*args) if method else None
[Wed Feb 16 15:01:29.524121 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]   File "/usr/lib/python3/dist-packages/freppledb/execute/management/commands/scenario_copy.py", line 437, in getHTML
[Wed Feb 16 15:01:29.524125 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438]     copy_perm = []
[Wed Feb 16 15:01:29.524129 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438] FileNotFoundError: [Errno 2] No such file or directory: '/usr/lib/python3/dist-packages/logs'

please help me to sort this out.

Thank You for your support.
S. Thipakar

**Comment by jdetaeye** [comment]
>[Wed Feb 16 15:01:29.524129 2022] [wsgi:error] [pid 35982:tid 139950161708800] [remote 123.231.127.183:30438] FileNotFoundError: [Errno 2] No such file or directory: '/usr/lib/python3/dist-packages/logs'

In a normally install this goes the log directory /var/log/frepple.
Only in development mode is the log folder under the project folder.
I don't know you installed frepple, but it doesn't look like a clean and proper install of a debian package.

Try creating the /var/log/frepple folder manually, and/or making it writeable for the os-user running the apache web server.
